src/sentry/api/endpoints/event_grouping_info.py

```diff
@@ -4,31 +4,26 @@
 
 from django.http import HttpResponse
 
-from sentry.api.base import Endpoint
-from sentry.api.bases.group import GroupPermission
+from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.exceptions import ResourceDoesNotExist
 from sentry.grouping.api import ConfigNotFoundException
 from sentry.models import Event
 from sentry.utils import json
 
 
-class EventGroupingInfoEndpoint(Endpoint):
-    permission_classes = (GroupPermission, )
-
-    def get(self, request, event_id):
+class EventGroupingInfoEndpoint(ProjectEndpoint):
+    def get(self, request, project, event_id):
         """
         Returns the grouping information for an event
         `````````````````````````````````````````````
 
         This endpoint returns a JSON dump of the metadata that went into the
         grouping algorithm.
         """
-        event = Event.objects.from_event_id(event_id, project_id=None)
+        event = Event.objects.from_event_id(event_id, project_id=project.id)
         if event is None:
             raise ResourceDoesNotExist
 
-        self.check_object_permissions(request, event.group)
```

Comment:
Why are we removing this permissions check? I guess just because we don't actually use the group in this api at all? Do we still need a permissions check for whether the user is alowed to access the event though?

Reply:
I checked the other endpoints located under this `/projects/:orgId/:projectId/events/:eventId/...` path and they all use the project permissions, I think it might be better to make this one the same

Reply:
Ah ok, makes sense. I guess this one just inherits an automatic project permisssions check from `ProjectEndpoint`?

Reply:
Yup